docs/spec.md

```diff
@@ -405,3 +464,26 @@ All integer types explicitly specified (uint32, uint64, etc.) in this section ar
 | ----- | ------ | ----------------------- |
 | 1     | opcode | 0x02                    |
 | 8     | uint64 | timestamp (nanoseconds) |
+
+#### Service Call Request
+
+- Request to call a service that has been advertised by the server.
+- Only supported if the server previously declared the `services` [capability](#server-info).
+
+| Bytes           | Type    | Description                                                                                  |
+| --------------- | ------- | -------------------------------------------------------------------------------------------- |
+| 1               | opcode  | 0x02                                                                                         |
+| 4               | uint32  | service id                                                                                   |
+| 8               | uint64  | call id, a unique number to identify the corresponding service response                      |
+| 4               | uint32  | id of the request encoding, as specified in the [service advertisement](#advertise-services) |
```

Comment:
The id above makes more sense now, to avoid a length-prefixed string here. I have a preference for a length-prefixed string instead of introducing the `encoding_id` concept.

Reply:
What length would you suggest? Or you mean to have a second field with the length of the encoding string?

Alternatively, we could make `acceptedEncoding` a string array and refer to the index here?